front/src/applications/operationalStudies/components/ManageTrainSchedule/Itinerary/DisplayItinerary/Destination.tsx

```diff
@@ -21,20 +22,58 @@ import { makeEnumBooleans } from 'utils/constants';
 
 import InputSNCF from 'common/BootstrapSNCF/InputSNCF';
 import { store } from 'Store';
-import { MODES, STDCM_MODES } from 'applications/operationalStudies/consts';
+import {
+  DEFAULT_MODE,
+  DEFAULT_STDCM_MODE,
+  MODES,
+  PointOnMap,
+  STDCM_MODES,
+} from 'applications/operationalStudies/consts';
+import { noop } from 'lodash';
 
 interface DestinationProps {
   zoomToFeaturePoint: (lngLat?: Position, id?: string, source?: string) => void;
+  stdcmMode?: symbol;
+  mode?: string;
+  destination?: PointOnMap | undefined;
+  destinationDate?: string | undefined;
+  destinationTime?: string | undefined;
+  dispatch?: Dispatch;
+}
+
+export function withOSRDData<T>(Component: ComponentType<T>) {
+  return (hocProps: T) => {
+    const stdcmMode = useSelector(getStdcmMode);
+    const mode = useSelector(getMode);
+    const destination = useSelector(getDestination);
+    const destinationDate = useSelector(getDestinationDate);
+    const destinationTime = useSelector(getDestinationTime);
+    const dispatch = useDispatch();
```

Comment:
Why do move dispatch out the the `Destination` component ?

Reply:
because the component is supposed to be able to work with no store (storybook)